src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -16,15 +16,21 @@ internal static partial class cgroups
     {
         // For cgroup v1, see https://www.kernel.org/doc/Documentation/cgroup-v1/
         // For cgroup v2, see https://www.kernel.org/doc/Documentation/cgroup-v2.txt
+        // For disambiguation, see https://systemd.io/CGROUP_DELEGATION/#three-different-tree-setups-
 
-        /// <summary>The version of cgroup that's being used </summary>
+        /// <summary>The supported versions of cgroup.</summary>
         internal enum CGroupVersion { None, CGroup1, CGroup2 };
 
+        /// <summary>Path to cgroup filesystem that tells us which version of cgroup is in use.</summary>
+        private const string SysFsCgroupFileSystemPath = "/sys/fs/cgroup";
         /// <summary>Path to mountinfo file in procfs for the current process.</summary>
         private const string ProcMountInfoFilePath = "/proc/self/mountinfo";
         /// <summary>Path to cgroup directory in procfs for the current process.</summary>
         private const string ProcCGroupFilePath = "/proc/self/cgroup";
 
+        /// <summary>The version of cgroup that's being used. Mutated by tests only.</summary>
+        internal static CGroupVersion s_cgroupVersion = FindCGroupVersion();
```

Comment:
```suggestion
        internal static readonly CGroupVersion s_cgroupVersion = FindCGroupVersion();
```

Reply:
Ah, you it's not readonly because it's "Mutated by tests"?  Is there a way to avoid that?

Reply:
Yes, there's definitely a way to do it. Please let me know if this new version where `s_cgroupVersion` is explicitly passed around looks better.